strapi/strapi
# Feat(blocks): move blocks

### What does it do?

- Adds the ability to move the blocks around in editor.
- Blocks and list-items are draggable, list block, link is not draggable.
- Keyboard shortcut Cmd/Ctrl + Shift + Up/Down ( referenced from notion) added to move blocks. 
- Quote and list styles updated as per design

`useDragAndDrop` hook Changes :  

1. Hook is updated to accept indices as `Array<number>`, to support nested list items path, for eg: [1, 0] or  for list-item path [1, 0, 0] and handled it separately.
2. Default functionality of moving items onMoveItem() updated and made it optional as for Blocks this needs to be handled onDropItem().


### How to test it?

On hover user should be able to see drag option and be able to drag block and list-items. While dragging user will see a blue line drop-placeholder on droppable area. On drop, item will be moved to respective position specified by drop placeholder.

[Design](https://www.figma.com/file/MWPKHNT98LLWHGRp3MyWqu/Blocks?type=design&node-id=1333-7753&mode=design&t=5pfwVeyaeGqOmwLY-0)

### Related issue(s)/PR(s)

Previous PR : https://github.com/strapi/strapi/pull/18690
